spotty/spotty/Scenes/Core/PlaylistScene/PlaylistViewController.swift

```diff
@@ -7,101 +7,81 @@
+        view.addSubview(tableView)
     }
     
     private func registerNibs() {
-        collectionView.register(PlaylistCell.self, forCellWithReuseIdentifier: PlaylistCell.reuseIdentifier)
+        tableView.register(TableLargeCell.self)
     }
     
     private func createDataSource() {
-        dataSource = UICollectionViewDiffableDataSource<Section, AnyHashable>(collectionView: collectionView) {
-            collectionView, indexPath, item in
+        dataSource = TableViewDataSource(tableView: tableView) {
+            tableView, indexPath, item in
             
-            let section = self.sections[indexPath.section]
-            return self.makeConfiguredCell(for: section,
-                                           collectionView: collectionView,
-                                           item: item,
-                                           indexPath: indexPath)
-        }
-    }
-    
-    private func makeConfiguredCell(for section: Section,
-                                    collectionView: UICollectionView,
-                                    item: AnyHashable,
-                                    indexPath: IndexPath) -> UICollectionViewCell {
-        switch section {
-        case .playlists:
-            return collectionView.configureReuseableCell(
-                PlaylistCell.self,
-                modelType: Playlist.self,
-                item: item,
-                indexPath: indexPath)
+            guard let section = Section.init(rawValue: indexPath.section),
+                  let cell = tableView.configuredReuseableCell(
+                    section.cellTypeReuseIdentifier,
+                    item: item,
+                    indexPath: indexPath) as? UITableViewCell
+            else { fatalError("Failed to create Reuseable Cell") }
+            
+            return cell
         }
     }
     
     private func bind() {
-        viewModel.playlists.bindAndFire { [weak self] playlists in
-            if let playlists = playlists { self?.reloadData(playlists: playlists) }
-        }
+        viewModel.playlists.bindAndFire { [weak self] _ in self?.reloadData() }
     }
     
-    private func reloadData(playlists: [Playlist]) {
-        var snapshot = NSDiffableDataSourceSnapshot<Section, AnyHashable>()
-        
-        snapshot.appendSections(sections)
+    private func reloadData() {
+        guard let playlists = viewModel.playlists.value else { return }
+        var snapshot = Snapshot()
+        snapshot.appendSections(Section.allCases)
         snapshot.appendItems(playlists, toSection: .playlists)
-        
         dataSource?.apply(snapshot)
     }
 }
 
-fileprivate enum Section {
+extension PlaylistViewController {
+    
+    static func create(viewModel: PlaylistViewModelProtocol = PlaylistViewModel()) -> UIViewController {
+        let viewController = PlaylistViewController()
+        viewController.viewModel = viewModel
+        viewController.title = Constant.SceneTitle.playlist
```

Comment:
You can get the title from the viewModel.

Reply:
Also set the value in the setupUI or bind methods as the part of the UI configuration.